### PR TITLE
Add YDKB Grape PCB

### PR DESCRIPTION
**Pull request by somepin**
## Description

Add QMK support for YDKB Grape PCB with default and VIA keymaps. ~~Attempted to use the SN74x138 demultiplexer driver (#16217) at first, but some keys didn't work. Ended up using the full custom matrix from this [TMK fork](https://github.com/yangdigi/tmk_keyboard_custom/blob/master/keyboard/Grape11900/matrix.c) instead, with the required quantum calls and callback defaults added.~~

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ydkb/grape/matrix.c [hunk 1]
```diff
@@ -0,0 +1,198 @@
+/*
```
Comment (drashna) [inline keyboards/ydkb/grape/matrix.c:1]: why does this have a custom matrix?

File: keyboards/ydkb/grape/readme.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# YDKB Grape
+
+![YDKB Grape](https://i.imgur.com/TZjgC9y.jpg)
```
Comment (drashna) [inline keyboards/ydkb/grape/readme.md:3]:
```suggestion
![YDKB Grape](https://i.imgur.com/TZjgC9yh.jpg)
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/ydkb/grape/matrix.c (hunk 1 above)
Comment (somepin) [reply keyboards/ydkb/grape/matrix.c:1]: The matrix rows are driven by a 74HC138 demultiplexer. My attempt to use the SN74x138 driver like the Wasdat, Wasdat Code, and AL1 does resulted in some keys not working. The existing custom matrix from the TMK fork that ydkb.io uses is fully functional.

Comment (drashna) [reply keyboards/ydkb/grape/matrix.c:1]: well, at the very least, this needs to be converted to the lite version of custom matrix, as zvecr indicated, and gpio commands should be used, not avr specifc ones.

Comment (somepin) [reply keyboards/ydkb/grape/matrix.c:1]: I was able to fix the lite version of the custom matrix with the SN74x138 driver, and all keys work now :)

**Review by drashna — APPROVED** [review]

File: keyboards/ydkb/grape/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_all( \
```
Comment (fauxpark) [inline keyboards/ydkb/grape/keymaps/via/keymap.c:19]:
```suggestion
    [0] = LAYOUT_all(
```

File: keyboards/ydkb/grape/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                                                                                                                                              KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
+        KC_GRV,  KC_1,    KC_2,    KC_3,   KC_4,   KC_5,   KC_6,   KC_7,   KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_BSPC,     KC_P1,   KC_P2,   KC_P3,   KC_PENT,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,   KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,              KC_P0,   KC_P0,   KC_PDOT, KC_PENT,
+        KC_CAPS, KC_A,    KC_S,    KC_D,   KC_F,   KC_G,   KC_H,   KC_J,   KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,                   KC_F13,   KC_F14,   KC_F15,   
+        KC_LSFT, KC_LSFT, KC_Z,    KC_X,   KC_C,   KC_V,   KC_B,   KC_N,   KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_RSFT, KC_UP,           KC_F16,   KC_F17,   KC_F18,   
+        KC_LCTL, KC_LGUI, KC_LALT,                 KC_SPC, KC_SPC,         KC_RALT, KC_RGUI, KC_APP,  KC_RCTL,          KC_LEFT, KC_DOWN, KC_RGHT,     KC_BTN1, KC_BTN2    
+    ),
+
+    [1] = LAYOUT_all( \
```
Comment (fauxpark) [inline keyboards/ydkb/grape/keymaps/via/keymap.c:30]:
```suggestion
    [1] = LAYOUT_all(
```

File: keyboards/ydkb/grape/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+                                                                                                                                                   _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,     _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,              _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,                _______,   _______,   _______,   
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,       _______,   _______,   _______,   
+        _______, _______, _______,               _______, _______,         _______, _______, _______,  _______,              _______, _______, _______,      _______, _______    
+    ),
+
+    [2] = LAYOUT_all( \
```
Comment (fauxpark) [inline keyboards/ydkb/grape/keymaps/via/keymap.c:41]:
```suggestion
    [2] = LAYOUT_all(
```

File: keyboards/ydkb/grape/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_all( \
```
Comment (fauxpark) [inline keyboards/ydkb/grape/keymaps/default/keymap.c:19]:
```suggestion
    [0] = LAYOUT_all(
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**